src/local/room_position.rs

```diff
@@ -0,0 +1,375 @@
+            "out of bounds room_x: {}",
+            room_name.x_coord,
+        );
+        assert!(
+            (-128..=127).contains(&room_name.y_coord),
+            "out of bounds room_y: {}",
+            room_name.y_coord,
+        );
+        let room_x = (room_name.x_coord + HALF_WORLD_SIZE) as u32;
+        let room_y = (room_name.y_coord + HALF_WORLD_SIZE) as u32;
+
+        LocalRoomPosition {
+            packed: (room_x << 24) | (room_y << 16) | (x << 8) | y,
+        }
+    }
+
+    #[inline]
+    pub fn packed_repr(self) -> i32 {
+        self.packed as i32
+    }
+
+    #[inline]
+    pub fn from_packed(packed: i32) -> Self {
+        LocalRoomPosition {
+            packed: packed as u32,
+        }
+    }
+
+    #[inline]
+    fn room_x(self) -> i32 {
+        (self.packed >> 24 & 0xFF) as i32 - HALF_WORLD_SIZE
+    }
+
+    #[inline]
+    fn room_y(self) -> i32 {
+        (self.packed >> 16 & 0xFF) as i32 - HALF_WORLD_SIZE
+    }
+
+    #[inline]
+    pub fn x(self) -> u32 {
+        self.packed >> 8 & 0xFF
+    }
+
+    #[inline]
+    pub fn y(self) -> u32 {
+        self.packed & 0xFF
+    }
+
+    #[inline]
+    pub fn room_name(self) -> LocalRoomName {
+        LocalRoomName {
+            x_coord: self.room_x(),
+            y_coord: self.room_y(),
+        }
+    }
+
+    #[inline]
+    pub fn set_x(&mut self, x: u32) {
+        assert!(x < 50, "out of bounds x: {}", x);
+        self.packed = (self.packed & !(0xFF << 8)) | (x << 8);
+    }
+
+    #[inline]
+    pub fn set_y(&mut self, y: u32) {
+        assert!(y < 50, "out of bounds y: {}", y);
+        self.packed = (self.packed & !0xFF) | y;
+    }
+
+    #[inline]
+    pub fn set_room_name(&mut self, room_name: LocalRoomName) {
+        assert!(
+            (-128..=127).contains(&room_name.x_coord),
+            "out of bounds room_x: {}",
+            room_name.x_coord,
+        );
+        assert!(
+            (-128..=127).contains(&room_name.y_coord),
+            "out of bounds room_y: {}",
+            room_name.y_coord,
+        );
+        let room_x = (room_name.x_coord + HALF_WORLD_SIZE) as u32;
+        let room_y = (room_name.y_coord + HALF_WORLD_SIZE) as u32;
+
+        self.packed = (self.packed & 0xFFFF) | (room_x << 24) | (room_y << 16);
+    }
+
+    #[inline]
+    pub fn with_x(mut self, x: u32) -> Self {
+        self.set_x(x);
+        self
+    }
+
+    #[inline]
+    pub fn with_y(mut self, y: u32) -> Self {
+        self.set_y(y);
+        self
+    }
+
+    #[inline]
+    pub fn with_room_name(mut self, room_name: LocalRoomName) -> Self {
+        self.set_room_name(room_name);
+        self
+    }
+}
+
+mod stdweb {
+    use stdweb::{Reference, Value};
+
+    use crate::{
+        macros::*,
+        objects::{HasPosition, RoomPosition},
+        traits::{TryFrom, TryInto},
+    };
+
+    use super::LocalRoomPosition;
+
+    impl LocalRoomPosition {
+        pub fn remote(self) -> RoomPosition {
```

Comment:
Given it is implemented here in a private module, I doubt it'll get included in the docs.

Reply:
I guess it isn't too intuitive, but `impl Type` blocks are always applicable, no matter the location. Many of the utility methods are implemented in private modules as well?

Reply:
So they inherit the struct's visibility? Interesting (and easier for when you want to split the implementation like in this case).

Reply:
As I understand it, `impl` blocks are always applicable, no matter where they're located in submodules. It's even technically possible to write `impl` blocks within functions, and in there they behave exactly like anywhere else.

In this PR I've implemented most of the methods in similar private modules, just ones declared at the top of the file and written in separate files.